### ISSUE TITLE
Reduce code duplication in front end

### ISSUE THREAD

**Issue by chrisxkeith** [usdigitalresponse/usdr-gost#965]
_This depends on there first being comprehensive front end test automation, so keep at a low priority._

Example:
[Dashboard.vue](https://github.com/usdigitalresponse/usdr-gost/blob/_staging/packages/client/src/views/Dashboard.vue)
vs.
[UpcomingClosingDates.vue](https://github.com/usdigitalresponse/usdr-gost/blob/_staging/packages/client/src/views/UpcomingClosingDates.vue)
and
[RecentActivity.vue](https://github.com/usdigitalresponse/usdr-gost/blob/_staging/packages/client/src/views/RecentActivity.vue)

**Comment by TylerHendrickson** [comment]
Waiting on a larger redesign / FE refactoring effort.
